npm/ng-packs/packages/core/src/lib/providers/core-module-config.provider.ts

```diff
@@ -107,13 +107,14 @@ export function provideAbpCore(...features: CoreFeature<CoreFeatureKind>[]) {
       }),
     ),
     provideAppInitializer(() => {
-      getInitialData();
-      localeInitializer();
       inject(LocalizationService);
       inject(LocalStorageListenerService);
       inject(RoutesHandler);
-    }),
 
+      return (async (): Promise<void> => {
```

Comment:
We could actually use such implementation instead:

```ts
    provideAppInitializer(async () => {
      inject(LocalizationService);
      inject(LocalStorageListenerService);
      inject(RoutesHandler);
      await getInitialData();
    })
```

 
The direct async approach is cleaner and more readable since it eliminates unnecessary function wrapping.

It also follows Angular conventions as this is the standard pattern for the old APP_INITIALIZER, and reduces complexity because `provideAppInitializer` handles async functions natively.